python/tk_maya/panel_generation.py

```diff
@@ -108,6 +109,20 @@ def dock_panel(engine, shotgun_panel, title, new_panel):
             # Embed the Shotgun app panel into the Maya panel workspace control.
             build_workspace_control_ui(shotgun_panel_name)
 
+            # Use a workaround to force Maya 2017 to refresh the panel size.
+            try:
+                # Create a new empty workspace control tab.
+                name = cmds.workspaceControl(uuid.uuid4().hex,
+                                             tabToControl=(maya_panel_name, -1),  # -1 to append a new tab
+                                             uiScript="",
+                                             r=True)  # raise at the top of its workspace area
+                # Delete the empty workspace control.
+                cmds.deleteUI(name)
+                # Delete the empty workspace control state.
+                cmds.workspaceControlState(name, remove=True)
+            except:
+                pass
```

Comment:
Comments added:
```
# We encased this workaround in a try/except since we cannot be sure
# that it will still work without errors in future versions of Maya.
```